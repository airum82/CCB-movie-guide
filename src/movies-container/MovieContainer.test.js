import React from 'react';
import MovieContainer from './MoviesContainer';
import { shallow } from 'enzyme';

describe('MovieContainer', () => {
  let wrapper;
  const mockProps = {
    movies: [{ title: 'fake1'}, { title: 'fake2'}],
    viewMovie: jest.fn(),
    location: {
      pathname: '/now_playing'
    },
    formatDate: jest.fn()
  }

  beforeEach(() => {
    wrapper = shallow(<MovieContainer  {...mockProps} />)
  })

  //rendering
  it('should match snapshot upon render', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have class "movie-container-single" when pathname includes "movie"', () => {
    mockProps.location.pathname = '/movie/543455';
    wrapper = shallow(<MovieContainer {...mockProps} />)
    expect(wrapper.find('section').hasClass('movie-container-single')).toBe(true);
  })

  it('should not have class "movie-container-single" when pathname does not include "movie"', () => {
    mockProps.location.pathname = '/now_playing';
    wrapper = shallow(<MovieContainer {...mockProps} />)
    expect(wrapper.find('section').hasClass('movie-container-single')).toBe(false)
  })

  it('should have class "movie-container" when pathname does not include "movie"', () => {
    expect(wrapper.find('section').hasClass('movie-container')).toBe(true);
  })

  it('should not have class "movie-container" when pathname does include "movie"', () => {
    mockProps.location.pathname = '/movie/3453';
    wrapper = shallow(<MovieContainer {...mockProps} />)
    expect(wrapper.find('section').hasClass('movie-container')).toBe(false)
  })
})