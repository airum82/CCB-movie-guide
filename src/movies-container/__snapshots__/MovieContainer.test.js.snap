// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should match snapshot upon render 1`] = `
<section
  className="movie-container"
>
  <Movie
    key="0"
    location={
      Object {
        "pathname": "/now_playing",
      }
    }
    movie={
      Object {
        "title": "fake1",
      }
    }
    viewMovie={[MockFunction]}
  />
  <Movie
    key="1"
    location={
      Object {
        "pathname": "/now_playing",
      }
    }
    movie={
      Object {
        "title": "fake2",
      }
    }
    viewMovie={[MockFunction]}
  />
</section>
`;
